From ad8102f2e0ab93f104f5bafb6c844787ba2247dc Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 3 Feb 2023 10:18:30 -0500
Subject: [PATCH 7/7] LLVM 15 fix

The attribute should not be applied to return value.
Too lazy to figure out the proper fix.
---
 src/codegen.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/codegen.cpp b/src/codegen.cpp
index ce5a9aa539..c6c614200a 100644
--- a/src/codegen.cpp
+++ b/src/codegen.cpp
@@ -898,11 +898,7 @@ static const auto jl_alloc_obj_func = new JuliaFunction{
     [](LLVMContext &C) { return AttributeList::get(C,
             AttributeSet::get(C, makeArrayRef({Attribute::getWithAllocSizeArgs(C, 1, None)})), // returns %1 bytes
 
-            Attributes(C, {Attribute::NoAlias, Attribute::NonNull,
-#if JL_LLVM_VERSION >= 150000
-            Attribute::get(C, Attribute::AllocKind, AllocFnKind::Alloc | AllocFnKind::Uninitialized | AllocFnKind::Aligned),
-#endif
-            }),
+            Attributes(C, {Attribute::NoAlias, Attribute::NonNull}),
             None); },
 };
 static const auto jl_newbits_func = new JuliaFunction{
-- 
2.39.0

